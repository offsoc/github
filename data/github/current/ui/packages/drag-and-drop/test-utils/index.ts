import {fireEvent} from '@storybook/test'

/**
 * Simulate mouse drag and drop events in a storybook environment.
 */

// https://stackoverflow.com/a/53946549/1179377
function isElement(obj: unknown): obj is Element {
  if (typeof obj !== 'object') {
    return false
  }
  let prototypeStr
  let prototype
  do {
    prototype = Object.getPrototypeOf(obj)
    // to work in iframe
    prototypeStr = Object.prototype.toString.call(prototype)
    // '[object Document]' is used to detect document
    if (prototypeStr === '[object Element]' || prototypeStr === '[object Document]') {
      return true
    }
    obj = prototype
    // null is the terminal of object
  } while (prototype !== null)
  return false
}

function getElementClientCenter(element: Element) {
  const {left, top, width, height} = element.getBoundingClientRect()
  return {
    x: left + width / 2,
    y: top + height / 2,
  }
}

const getCoords = (element?: Element) => (isElement(element) ? getElementClientCenter(element) : element)

const sleep = (ms: number) =>
  new Promise(resolve => {
    setTimeout(resolve, ms)
  })

export async function dragEnd(element: Element, current: {clientX: number; clientY: number}): Promise<void> {
  await fireEvent.mouseUp(element, current)
}

export async function dragMove(
  element: Element,
  current: {clientX: number; clientY: number},
  step: {x: number; y: number},
  steps = 20,
  duration = 500,
): Promise<{clientX: number; clientY: number}> {
  for (let i = 0; i < steps; i++) {
    current.clientX += step.x
    current.clientY += step.y
    await sleep(duration / steps)
    await fireEvent.mouseMove(element, current)
  }
  return current
}

type DragStartParams = {
  element: Element
  delta?: {
    x: number
    y: number
  }
  to?: Element
  steps?: number
  duration?: number
}

export async function dragStart({
  element,
  to: inTo,
  delta,
  steps = 20,
}: DragStartParams): Promise<{currentPosition: {clientX: number; clientY: number}; mouseStep: {x: number; y: number}}> {
  const from = getElementClientCenter(element)
  const to = (
    delta
      ? {
          x: from.x + delta.x,
          y: from.y + delta.y,
        }
      : getCoords(inTo)
  ) as {x: number; y: number}

  const mouseStep = {
    x: (to.x - from.x) / steps,
    y: (to.y - from.y) / steps,
  }

  const currentPosition = {
    clientX: from.x,
    clientY: from.y,
  }

  await fireEvent.mouseEnter(element, currentPosition)
  await fireEvent.mouseOver(element, currentPosition)
  await fireEvent.mouseMove(element, currentPosition)
  await fireEvent.mouseDown(element, currentPosition)

  // Slight bump to trigger 'drag' state
  await fireEvent.mouseMove(element, {clientX: 0, clientY: 1})

  return {currentPosition, mouseStep}
}
