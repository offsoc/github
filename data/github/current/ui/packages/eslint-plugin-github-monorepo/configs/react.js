module.exports = {
  rules: {
    '@github-ui/github-monorepo/react-partial-name': 'error',
    // enable on a per-package basis as migrated
    '@github-ui/github-monorepo/no-sx': 'off',
  },
}
