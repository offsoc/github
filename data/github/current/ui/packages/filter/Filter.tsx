import {type TestIdProps, testIdProps} from '@github-ui/test-id-props'
import {Box, type SxProp, useResponsiveValue} from '@primer/react'
import type {ResponsiveValue} from '@primer/react/lib-esm/hooks/useResponsiveValue'
import type React from 'react'
import {useCallback, useState} from 'react'

import {AdvancedFilterDialog, type FilterButtonVariant} from './advanced-filter-dialog/AdvancedFilterDialog'
import {ClearButton} from './clear-button/ClearButton'
import {defaultFilterSettings} from './constants/defaults'
import {FilterContextProvider} from './context/RootContext'
import {FilterQuery} from './filter-query'
import {FilterInputIcon} from './FilterInputIcon'
import {FilterInputWrapper} from './FilterInputWrapper'
import {Input} from './input/Input'
import {SubmitButton} from './input/SubmitButton'
import {SuggestionsList} from './suggestions/SuggestionsList'
import type {FilterProvider, FilterSettings, FilterVariant, SubmitEvent} from './types'
import {ValidationMessage} from './ValidationMessage'

// Exports from the root of the package
export * from './constants/defaults'
export * from './filter-query'
export * from './FilterRevert'
export * from './types'
export * from './ValidationMessage'

const filterInputWrapperStyles = ({variant, onSubmit}: {variant: FilterProps['variant']; onSubmit: boolean}) => {
  const leftRounded = variant === 'input'
  const rightRounded = (variant === 'input' && !onSubmit) || (variant !== 'input' && !onSubmit)

  return {
    display: 'grid',
    gridTemplateColumns: onSubmit ? '1fr min-content' : 'min-content 1fr min-content',
    alignItems: 'center',
    pl: onSubmit ? '12px' : 1,
    pr: '1px',
    width: '100%',
    borderStyle: 'solid',
    borderWidth: '1px',
    borderColor: 'border.default',
    marginInlineEnd: -1,
    ...(leftRounded && {
      borderTopLeftRadius: 2,
      borderBottomLeftRadius: 2,
    }),
    ...(rightRounded && {
      borderTopRightRadius: 2,
      borderBottomRightRadius: 2,
    }),
    ':has(input:focus-visible)': {
      zIndex: 1,
      outline: '2px solid var(--borderColor-accent-emphasis, var(--color-accent-emphasis))',
      outlineOffset: '-2px',
    },
  }
}

export type FilterProps = {
  /** ID of the Filter Component */
  id: string
  /** Change event callback, passing the updated filter value */
  onChange?: (value: string) => void
  /** Event callback, triggered by the filter query being parsed, and passing the updated filter query request*/
  onParse?: (request: FilterQuery) => void
  /** Submit event callback, triggered by Enter or pressing Submit, and passing the updated filter query and the
   * originating event type */
  onSubmit?: (request: FilterQuery, eventType: SubmitEvent) => void
  /** Array of FilterProviders that are being used/supported */
  providers: FilterProvider[]
  /** Context to be applied for all API queries */
  context?: Record<string, string>
  /** Variant of the filter button, whether to show the title or just the icon */
  filterButtonVariant?: FilterButtonVariant
  /** Initial value to use for the input. DO NOT USE in conjuction with `filterValue`.
   * Only use this if you want the Filter to have an initial value but remain uncontrolled. */
  initialFilterValue?: string
  /** Supplied filter value to use. By using this, the Filter is externally controlled
   * and will not use it's own state. */
  filterValue?: string
  /** React Ref to the input element */
  inputRef?: React.RefObject<HTMLInputElement>
  /** Label displayed atop the Filter bar. This is only shown when `visuallyHideLabel` is `false` */
  label: string
  /** Input keydown event callback */
  onKeyDown?: React.KeyboardEventHandler<Element>
  /** Validation callback, passing the validation message and the validation query */
  onValidation?: (messages: string[], filterQuery: FilterQuery) => void
  /** Placeholder text for the input */
  placeholder?: string
  /** Settings for the Filter, such as support of the legacy No filter provider and alias matching */
  settings?: FilterSettings
  /** Whether to show any validation messages or not. If this is disabled, use `onValidation` to receive the messages to display elsewhere */
  showValidationMessage?: boolean
  /** Variant of the Filter to render. `button` or `input` for one or the other, `full` (default) renders both.  */
  variant?: FilterVariant
  /** Whether to visually hide the Label or not */
  visuallyHideLabel?: boolean
} & TestIdProps &
  SxProp

const defaultFilterQuery = new FilterQuery()

export function Filter({
  context,
  'data-testid': dataTestId,
  filterButtonVariant = 'normal',
  filterValue,
  id,
  initialFilterValue,
  inputRef,
  label,
  onChange,
  onParse,
  onSubmit,
  onKeyDown,
  onValidation,
  placeholder,
  providers,
  settings,
  showValidationMessage = true,
  sx,
  variant = 'full',
  visuallyHideLabel = true,
}: FilterProps) {
  const [validationMessage, setValidationMessage] = useState<string[]>([])
  const hasValidationMessage = showValidationMessage && validationMessage.length > 0

  const validationCallback = useCallback(
    (messages: string[] = [], filterQuery: FilterQuery = defaultFilterQuery) => {
      setValidationMessage(messages)
      onValidation?.(messages, filterQuery)
    },
    [onValidation],
  )

  const filterButtonResponsive = useResponsiveValue<ResponsiveValue<FilterButtonVariant>, FilterButtonVariant>(
    {regular: filterButtonVariant, narrow: variant === 'button' ? 'normal' : 'compact'},
    filterButtonVariant,
  )

  const externallyControlled = typeof filterValue === 'string'
  const [innerValue, setInnerValue] = useState<string>(initialFilterValue || '')

  const changeCallback = useCallback(
    (value: string) => {
      if (!externallyControlled) {
        setInnerValue(value)
      }
      onChange?.(value)
    },
    [externallyControlled, onChange],
  )

  return (
    <FilterContextProvider
      context={context}
      inputRef={inputRef}
      rawFilter={externallyControlled ? filterValue : innerValue}
      onChange={changeCallback}
      onParse={onParse}
      onSubmit={onSubmit}
      onValidation={validationCallback}
      providers={providers}
      settings={{...defaultFilterSettings, ...settings}}
      variant={variant}
    >
      <Box
        id={id}
        role="form"
        className="FormControl FormControl--fullWidth"
        sx={{...sx, gap: 0}}
        {...testIdProps(dataTestId ?? 'filter')}
      >
        <label
          id={`${id}-label`}
          htmlFor={`${id}-input`}
          className={`FormControl-label ${visuallyHideLabel ? 'sr-only' : ''}`}
          {...testIdProps('filter-bar-label')}
        >
          {label}
        </label>
        <Box sx={{display: 'grid', gap: 3, width: '100%', position: 'relative'}}>
          <FilterInputWrapper isStandalone={variant === 'button'}>
            {variant !== 'input' && (
              <AdvancedFilterDialog isStandalone={variant === 'button'} filterButtonVariant={filterButtonResponsive} />
            )}
            {variant !== 'button' && (
              <>
                <Box
                  sx={filterInputWrapperStyles({
                    variant,
                    onSubmit: !!onSubmit,
                  })}
                >
                  {!onSubmit && <FilterInputIcon />}
                  <Input
                    id={id}
                    placeholder={placeholder}
                    onKeyDown={onKeyDown}
                    hasValidationMessage={hasValidationMessage}
                  />
                  <ClearButton />
                </Box>
                {onSubmit && <SubmitButton />}
              </>
            )}
          </FilterInputWrapper>
          <SuggestionsList id={id} />
          {/* This needs to move outside the Filter component because we can't predict where the consumer wants to put the error message. In memex this would go above the table view. */}
          {showValidationMessage && <ValidationMessage messages={validationMessage} id={`${id}-validation-message`} />}
        </Box>
      </Box>
    </FilterContextProvider>
  )
}
