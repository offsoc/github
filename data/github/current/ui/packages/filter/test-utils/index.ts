import {act, fireEvent, screen} from '@testing-library/react'

export const setupExpectedAsyncErrorHandler = () => {
  jest.spyOn(console, 'error').mockImplementation((message: string) => {
    // * Because Filter is asynchronous, there are console errors that are thrown, but expected. This will rethrow
    // * any errors that are not related to the async nature of the component.
    if (!message.includes?.('wrapped in act(')) {
      // eslint-disable-next-line no-console
      console.warn(message)
    }
  })
}

export async function updateFilterValue(key: string, value: string = '') {
  jest.useFakeTimers()

  const input = screen.getByRole('combobox')
  input.focus()

  fireEvent.change(input, {target: {value: `${key}${key && value ? ':' : ''}${value}`}})

  // eslint-disable-next-line @typescript-eslint/require-await
  await act(async () => {
    jest.runAllTimers()
  })
}
