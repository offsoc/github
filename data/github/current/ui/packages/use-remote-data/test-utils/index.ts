import {act} from '@testing-library/react'

// https://github.com/tc39/proposal-promise-with-resolvers
export function withResolvers<R>() {
  let resolve: (value: R) => unknown
  let reject: (value?: unknown) => unknown
  const promise = new Promise<R>((res, rej) => {
    resolve = res
    reject = rej
  })

  // @ts-expect-error those are set sync by the time we get here
  return {resolve, reject, promise}
}

export async function nextTick() {
  return act(
    () =>
      new Promise(resolve => {
        setTimeout(resolve, 1)
      }),
  )
}
