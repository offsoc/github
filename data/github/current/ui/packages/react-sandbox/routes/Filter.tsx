import {Filter, ProviderSupportStatus, type FilterProvider, type FilterQuery} from '@github-ui/filter'
import {
  ArchivedFilterProvider,
  AssigneeFilterProvider,
  AuthorFilterProvider,
  ClosedFilterProvider,
  CommenterFilterProvider,
  CommentsFilterProvider,
  CreatedFilterProvider,
  DraftFilterProvider,
  InBodyFilterProvider,
  InCommentsFilterProvider,
  InTitleFilterProvider,
  InteractionsFilterProvider,
  InvolvesFilterProvider,
  IsFilterProvider,
  LabelFilterProvider,
  LanguageFilterProvider,
  LinkedFilterProvider,
  MentionsFilterProvider,
  MergedFilterProvider,
  MilestoneFilterProvider,
  OrgFilterProvider,
  ProjectFilterProvider,
  ReactionsFilterProvider,
  ReasonFilterProvider,
  RepositoryFilterProvider,
  ReviewFilterProvider,
  ReviewRequestedFilterProvider,
  ReviewedByFilterProvider,
  SortFilterProvider,
  StateFilterProvider,
  StatusFilterProvider,
  TypeFilterProvider,
  UpdatedFilterProvider,
  UserFilterProvider,
  UserReviewRequestedFilterProvider,
} from '@github-ui/filter/providers'
import {useState} from 'react'

export function FilterPage() {
  const [filterValue, setFilterValue] = useState('')
  const [submittedValue, setSubmittedValue] = useState('')

  const defaultUserObject = {
    currentUserLogin: 'monalisa',
    currentUserAvatarUrl: 'https://avatars.githubusercontent.com/u/90914?v=4',
  }

  const defaultProviders: FilterProvider[] = [
    new AssigneeFilterProvider(defaultUserObject),
    new AuthorFilterProvider(defaultUserObject, {filterTypes: {multiKey: false}}),
    new CommenterFilterProvider(defaultUserObject),
    new InvolvesFilterProvider(defaultUserObject),
    new MentionsFilterProvider(defaultUserObject),
    new ReviewedByFilterProvider(defaultUserObject),
    new ReviewRequestedFilterProvider(defaultUserObject),
    new UserFilterProvider(defaultUserObject),
    new UserReviewRequestedFilterProvider(defaultUserObject),
    new ArchivedFilterProvider(),
    new ClosedFilterProvider(),
    new CommentsFilterProvider(),
    new CreatedFilterProvider(),
    new DraftFilterProvider(),
    new InteractionsFilterProvider(),
    new IsFilterProvider(),
    new LinkedFilterProvider(),
    new MergedFilterProvider(),
    new StateFilterProvider('mixed', {
      filterTypes: {
        inclusive: true,
        exclusive: true,
        valueless: false,
        multiKey: false,
        multiValue: true,
      },
    }),
    new ReactionsFilterProvider(),
    new ReasonFilterProvider(),
    new ReviewFilterProvider(),
    new StatusFilterProvider({
      filterTypes: {
        inclusive: true,
        exclusive: true,
        valueless: true,
        multiKey: true,
        multiValue: false,
      },
    }),
    new TypeFilterProvider(),
    new LanguageFilterProvider({support: {status: ProviderSupportStatus.Deprecated}}),
    new UpdatedFilterProvider(),
    new ProjectFilterProvider(),
    new LabelFilterProvider(),
    new InBodyFilterProvider(),
    new InCommentsFilterProvider(),
    new InTitleFilterProvider(),
    new OrgFilterProvider(),
    new SortFilterProvider(['created', 'updated', 'reactions', 'comments', 'author-date', 'committer-date']),
    new MilestoneFilterProvider(),
    new RepositoryFilterProvider(),
  ]

  return (
    <div data-hpc>
      <Filter
        id="sandbox-filter"
        context={{repo: 'monalisa/smile'}}
        label="Filter items"
        filterValue={filterValue}
        providers={defaultProviders}
        onChange={(value: string) => setFilterValue(value)}
        onSubmit={(request: FilterQuery) => setSubmittedValue(request.raw)}
      />

      <div>{submittedValue}</div>
    </div>
  )
}
