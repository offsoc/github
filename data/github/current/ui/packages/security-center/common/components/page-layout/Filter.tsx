import {Filter as UiFilter, type FilterProvider} from '@github-ui/filter'
import {useEffect, useState} from 'react'

type Props = {
  providers: FilterProvider[]
  query: string
  onSubmit: (query: string) => void
}

function Filter({providers, query, onSubmit}: Props): JSX.Element {
  // state of the filter input control, before it is submitted
  const [filterValue, setFilterValue] = useState(query)

  // if the prop changes externally, update our state (e.g. on filter revert)
  useEffect(() => {
    setFilterValue(query)
  }, [query])

  return (
    <UiFilter
      id="security-overview-page-filter"
      label="Filter"
      placeholder="Filter"
      sx={{flexGrow: 1}}
      providers={providers}
      filterValue={filterValue}
      onChange={setFilterValue}
      onSubmit={request => onSubmit(request.raw)}
    />
  )
}

Filter.displayName = 'PageLayout.Filter'

export default Filter
