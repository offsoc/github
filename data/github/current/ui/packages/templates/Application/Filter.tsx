import {Box} from '@primer/react'
import {useState} from 'react'
import {Filter, type FilterProvider} from '@github-ui/filter'

import {AssigneeFilterProvider, AuthorFilterProvider, UserFilterProvider} from '@github-ui/filter/providers'

export default function ExampleFilter() {
  const [filterValue, setFilterValue] = useState('assignee:@me')
  return (
    <Box sx={{display: 'flex', width: '100%', flexDirection: 'column'}}>
      <Filter
        id="storybook-filter"
        label="Filter items"
        filterValue={filterValue}
        providers={defaultProviders}
        onChange={(value: string) => setFilterValue(value)}
      />
    </Box>
  )
}

const defaultUserObject = {
  currentUserLogin: 'monalisa',
  currentUserAvatarUrl: 'https://avatars.githubusercontent.com/u/90914?v=4',
}

const defaultProviders: FilterProvider[] = [
  new AssigneeFilterProvider(defaultUserObject),
  new AuthorFilterProvider(defaultUserObject, {filterTypes: {multiKey: false}}),
  new UserFilterProvider(defaultUserObject),
]
