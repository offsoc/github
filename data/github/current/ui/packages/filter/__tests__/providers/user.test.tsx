import {render, screen, waitFor} from '@testing-library/react'

import {Filter} from '../../Filter'
import {
  AssigneeFilterProvider,
  AuthorFilterProvider,
  CommenterFilterProvider,
  InvolvesFilterProvider,
  MentionsFilterProvider,
  ReviewedByFilterProvider,
  ReviewRequestedFilterProvider,
  UserFilterProvider,
  UserReviewRequestedFilterProvider,
} from '../../providers'
import {updateFilterValue} from '../../test-utils'
import {
  appendToFilterAndRenderAsyncSuggestions,
  expectFilterValueToBe,
  expectSuggestionsToMatchSnapshot,
  moveCursor,
  selectSuggestion,
  setupAsyncErrorHandler,
  setupUsersMockApi,
} from '../utils/helpers'

describe('Assignee', () => {
  setupAsyncErrorHandler()
  setupUsersMockApi()

  it('ensures `signed-in user` is part of accessible name for `@me` value', async () => {
    const filterProviders = [new AssigneeFilterProvider({showAtMe: true})]
    render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

    await updateFilterValue('assignee:')

    expect(screen.getByRole('option', {name: '@me, Signed-in user'})).toBeInTheDocument()
  })

  it('should filter and select suggestions based on name', async () => {
    const filterProviders = [new AssigneeFilterProvider({showAtMe: true})]
    render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

    await updateFilterValue('assignee:dusa')

    expectSuggestionsToMatchSnapshot()

    await selectSuggestion('dusave')

    expectFilterValueToBe('assignee:dusave')
  })

  it('should not show any suggestions based on invalid name', async () => {
    const filterProviders = [new AssigneeFilterProvider({showAtMe: true})]
    render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

    await updateFilterValue('duuuuuus')

    await waitFor(() => {
      expect(screen.getByTestId('filter-results')).toBeEmptyDOMElement()
    })
  })

  it('should filter and select suggestions when added in the start of multiple values', async () => {
    const filterProviders = [new AssigneeFilterProvider({showAtMe: true})]
    render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

    await updateFilterValue('is:issue assignee:,2percentsilk text')

    moveCursor('is:issue assignee:'.length)

    await appendToFilterAndRenderAsyncSuggestions('dus')

    expectSuggestionsToMatchSnapshot()

    await selectSuggestion('dusave')

    expectFilterValueToBe('is:issue assignee:dusave,2percentsilk text')
  })
})

describe('Author', () => {
  setupAsyncErrorHandler()
  setupUsersMockApi()

  it('should filter and select suggestions based on name', async () => {
    const filterProviders = [new AuthorFilterProvider({showAtMe: true})]
    render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

    await updateFilterValue('author:dusa')

    expectSuggestionsToMatchSnapshot()

    await selectSuggestion('dusave')

    expectFilterValueToBe('author:dusave')
  })

  it('should not show any suggestions based on invalid name', async () => {
    const filterProviders = [new AuthorFilterProvider({showAtMe: true})]
    render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

    await updateFilterValue('duuuuuus')

    await waitFor(() => {
      expect(screen.getByTestId('filter-results')).toBeEmptyDOMElement()
    })
  })
})

describe('Commenter', () => {
  setupAsyncErrorHandler()
  setupUsersMockApi()

  it('should filter and select suggestions based on name', async () => {
    const filterProviders = [new CommenterFilterProvider({showAtMe: true})]
    render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

    await updateFilterValue('commenter:dusa')

    expectSuggestionsToMatchSnapshot()

    await selectSuggestion('dusave')

    expectFilterValueToBe('commenter:dusave')
  })
})

describe('Involves', () => {
  setupAsyncErrorHandler()
  setupUsersMockApi()

  it('should filter and select suggestions based on name', async () => {
    const filterProviders = [new InvolvesFilterProvider({showAtMe: true})]
    render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

    await updateFilterValue('involves:dusa')

    expectSuggestionsToMatchSnapshot()

    await selectSuggestion('dusave')

    expectFilterValueToBe('involves:dusave')
  })
})

describe('Mentions', () => {
  setupAsyncErrorHandler()
  setupUsersMockApi()

  it('should filter and select suggestions based on name', async () => {
    const filterProviders = [new MentionsFilterProvider({showAtMe: true})]
    render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

    await updateFilterValue('mentions:dusa')

    expectSuggestionsToMatchSnapshot()

    await selectSuggestion('dusave')

    expectFilterValueToBe('mentions:dusave')
  })
})

describe('ReviewedBy', () => {
  setupAsyncErrorHandler()
  setupUsersMockApi()

  it('should filter and select suggestions based on name', async () => {
    const filterProviders = [new ReviewedByFilterProvider({showAtMe: true})]
    render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

    await updateFilterValue('reviewed-by:dusa')

    expectSuggestionsToMatchSnapshot()

    await selectSuggestion('dusave')

    expectFilterValueToBe('reviewed-by:dusave')
  })
})

describe('ReviewedRequested', () => {
  setupAsyncErrorHandler()
  setupUsersMockApi()

  it('should filter and select suggestions based on name', async () => {
    const filterProviders = [new ReviewRequestedFilterProvider({showAtMe: true})]
    render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

    await updateFilterValue('review-requested:dusa')

    expectSuggestionsToMatchSnapshot()

    await selectSuggestion('dusave')

    expectFilterValueToBe('review-requested:dusave')
  })
})

describe('User', () => {
  setupAsyncErrorHandler()
  setupUsersMockApi()

  it('should filter and select suggestions based on name', async () => {
    const filterProviders = [new UserFilterProvider({showAtMe: true})]
    render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

    await updateFilterValue('user:dusa')

    expectSuggestionsToMatchSnapshot()

    await selectSuggestion('dusave')

    expectFilterValueToBe('user:dusave')
  })
})

describe('UserReviewRequested', () => {
  setupAsyncErrorHandler()
  setupUsersMockApi()

  it('should filter and select suggestions based on name', async () => {
    const filterProviders = [new UserReviewRequestedFilterProvider({showAtMe: true})]
    render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

    await updateFilterValue('user-review-requested:dusa')

    expectSuggestionsToMatchSnapshot()

    await selectSuggestion('dusave')

    expectFilterValueToBe('user-review-requested:dusave')
  })
})
