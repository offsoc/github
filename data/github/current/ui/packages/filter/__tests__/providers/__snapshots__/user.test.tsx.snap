// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assignee should filter and select suggestions based on name 1`] = `
[
  "dusaveDustin Savery",
  "evilDusaveEvil Dustin Savery",
]
`;

exports[`Assignee should filter and select suggestions when added in the start of multiple values 1`] = `
[
  "dusaveDustin Savery",
  "evilDusaveEvil Dustin Savery",
]
`;

exports[`Author should filter and select suggestions based on name 1`] = `
[
  "dusaveDustin Savery",
  "evilDusaveEvil Dustin Savery",
]
`;

exports[`Commenter should filter and select suggestions based on name 1`] = `
[
  "dusaveDustin Savery",
  "evilDusaveEvil Dustin Savery",
]
`;

exports[`Involves should filter and select suggestions based on name 1`] = `
[
  "dusaveDustin Savery",
  "evilDusaveEvil Dustin Savery",
]
`;

exports[`Mentions should filter and select suggestions based on name 1`] = `
[
  "dusaveDustin Savery",
  "evilDusaveEvil Dustin Savery",
]
`;

exports[`ReviewedBy should filter and select suggestions based on name 1`] = `
[
  "dusaveDustin Savery",
  "evilDusaveEvil Dustin Savery",
]
`;

exports[`ReviewedRequested should filter and select suggestions based on name 1`] = `
[
  "dusaveDustin Savery",
  "evilDusaveEvil Dustin Savery",
]
`;

exports[`User should filter and select suggestions based on name 1`] = `
[
  "dusaveDustin Savery",
  "evilDusaveEvil Dustin Savery",
]
`;

exports[`UserReviewRequested should filter and select suggestions based on name 1`] = `
[
  "dusaveDustin Savery",
  "evilDusaveEvil Dustin Savery",
]
`;
